tokio/tests/rt_basic.rs

```diff
@@ -58,15 +62,18 @@ fn no_extra_poll() {
         }
     }
 
-    let (tx, rx) = mpsc::unbounded_channel();
-    let mut rx = TrackPolls {
+    let (tx, rx) = support::mpsc_stream::unbounded_channel_stream::<()>();
+    let rx = TrackPolls {
         npolls: Arc::new(AtomicUsize::new(0)),
         s: rx,
     };
     let npolls = Arc::clone(&rx.npolls);
 
     let rt = rt();
 
+    // TODO: could probably avoid this, but why not.
+    let mut rx = Box::pin(rx);
+
     rt.spawn(async move { while rx.next().await.is_some() {} });
```

Comment:
To avoid it, you could use `tokio::pin!`, but it would need to be inside the spawned task. Not that it matters much as it is a test.